synapse/lib/stormtypes.py

```diff
@@ -2937,6 +2951,15 @@ def __str__(self):
     def __len__(self):
         return len(self.valu)
 
+    def __hash__(self):
+        # As a note, this hash of the typename and the value means that s_stormtypes.Str('foo') != 'foo'
```

Comment:
Why do we make the hash of `(typename, self.valu)` rather than just `self.valu`?

Reply:
After talking with epiphyte, I figured that was a little more type safe that just hashing the straight value. I tend to  treading carefully through the type system minefield though.

As a related sidenote: It certainly doesn't fix the problem of something like `s_stormtypes.Str("foo")` not equaling the straight string `foo`, but it doesn't make it worse, and that's really not something that's going to be fixed until we get more Prim's as the only thing in the storm runtime away.